toqito/channel_metrics/channel_distinguishability.py

```diff
@@ -0,0 +1,182 @@
+        from toqito.channel_metrics import channel_distinguishability
+        # Define two amplitude damping channels with gamma=0.25 and gamma=0.5
+        choi_ch_1 = kraus_to_choi(amplitude_damping(gamma=0.25))
+        choi_ch_2 = kraus_to_choi(amplitude_damping(gamma=0.5))
+
+        channel_distinguishability(choi_ch_1, choi_ch_2, None, [2, 2], strategy="minimax",
+                        primal_dual="primal")
+
+
+    References
+    ==========
+    .. footbibliography::
+        :filter: docname in docnames
+
+
+    :raises ValueError: If channels have different input or output dimensions.
+    :raises ValueError: If prior probabilities do not add up to 1.
+    :raises ValueError: If number of prior probabilities not equal to 2.
+    :param phi: A superoperator. It should be provided either as a Choi matrix,
+                or as a (1d or 2d) list of numpy arrays whose entries are its Kraus operators.
+    :param psi: A superoperator. It should be provided either as a Choi matrix,
+                or as a (1d or 2d) list of numpy arrays whose entries are its Kraus operators.
+    :param p: Prior probabilities of the two channels.
+    :param dim: Input and output dimensions of the channels.
+    :param strategy: Whether to perform Bayesian or minimax discrimination task. Possible
+                     values are "Bayesian" and "minimax". Defualt option is `strategy="Bayesian"`.
+    :param solver: Optimization option for `picos` solver. Default option is `solver="cvxopt"`.
+    :param primal_dual: Option for the optimization problem. Defualt option is `solver="cvxopt"`.
+    :param kwargs: Additional arguments to pass to picos' solve method.
+    :return: The optimal probability of discriminating two quantum channels.
+
+    """
+    # Get the input, output and environment dimensions of phi and psi.
+    d_in_phi, d_out_phi, d_e = channel_dim(phi, dim)
+    d_in_psi, d_out_psi, d_e = channel_dim(psi, dim)
+
+    # If the variable `phi` and/or `psi` are provided as a list, we assume this is a list
+    # of Kraus operators. We convert to choi matrices if not provided as choi matrix.
+    if isinstance(phi, list):
+        phi = kraus_to_choi(phi)
+
+    if isinstance(psi, list):
+        psi = kraus_to_choi(psi)
+
+    dim_phi, dim_psi = np.array([d_in_phi, d_out_phi]), np.array([d_in_psi, d_out_psi])
+
+    # checking for errors.
+    if not np.array_equal(dim_phi, dim_psi):
+        raise ValueError("The channels must have the same dimension input and output spaces as each other.")
+
+    if strategy.lower() == "bayesian":
+        if len(p) != 2:
+            raise ValueError("p must be a probability distribution with 2 entries.")
+
+        if max(p) >= 1:
+            return 1
+
+        if abs(sum(p) - 1) != 0:
+            raise ValueError("Sum of prior probabilities must add up to 1.")
+
+        return 1 / 2 * (1 + completely_bounded_trace_norm(p[0] * phi - p[1] * psi))
+
+    elif strategy.lower() == "minimax":
```

Comment:
based on the pytest coverage report, there is something wrong with this condition for the code block.